### DISCUSSION TITLE
Using the balance scheme for linear analysis

### DISCUSSION THREAD

**Discussion by jeankpf** [idefix-code/idefix#254] [Q&A]
Hi all,

I know that a specific branch balanceScheme is available in IDEFIX to impose an initial condition to be an equilibrium state, by computing initial residuals and removing them at each step. But I never used it myself, so I'm opening this discussion to have feedback from people who already did. At the moment, I can't compile it, so I'm wondering if there is a specific option to activate in the CMakeLists.txt.

Cheers,
Jean

**Comment by jeankpf** [comment]
Just in case anyone interested stumble upon this discussion in the future: an updated branch has been created on the official repo of the IDEFIX code (under the name updatedBalanceScheme) and is working (but not intended to be merged with the develop/master branch). I copied this branch on my own public IDEFIX fork (under the name myBalanceScheme), and I'm trying to keep it updated with new features of the develop and/or other branches. For instance, my branch currently support python-binding, so feel free to use it if needed.

**Reply by glesur** [reply]
Thanks @jeankpf for keeping this branch alive.

For reference: this branch is not planned to be merged in the main repository because it is very tricky to use properly the balanced scheme. The technic used is essentially to measure the righthand side numerical residual starting from an analytical equilibrium. We then substract this residual in the subsequent time evolution at each timestep.

In principle, this guarantees that the analytical equilibrium is recovered at machine precision, but it doesn't formerly guarantee that perturbation will necessarily follow the linear operator around said equilibrium. It is very likely it will do so, but there might be situations where it is not the case.

Worse, it you code a state that is not an exact equilibrium, the resulting evolution will be incorrect, but it will be hard to notice.

So to avoid any big mistake, I prefer to leave this branch alone, unless there is a very strong push for this feature to be explicitly supported.
